src/d3d11/d3d11_context_ext.cpp

```diff
@@ -133,5 +139,76 @@ namespace dxvk {
       ctx->setBarrierControl(cFlags);
     });
   }
-  
+
+
+  class CubinShaderLaunchInfo {
+  public:
+    Com<CubinShaderWrapper> cuShader;
+    std::vector<uint8_t> Params;
+    size_t ParamSize;
+    VkCuLaunchInfoNVX nvxLaunchInfo = { VK_STRUCTURE_TYPE_CU_LAUNCH_INFO_NVX };
+    const void* CuLaunchConfig[5];
+    std::vector<std::tuple<Rc<DxvkBuffer>, bool /*read*/, bool /*write*/>> buffers;
+    std::vector<std::tuple<Rc<DxvkImage>, bool /*read*/, bool /*write*/>> images;
```

Comment:
Could use `DxvkAccessFlags` instead of two bools for clarity (supported bits are `DxvkAccess::Read and `DxvkAccess::Write`, respectively). It's a bit more clunky than it should be though (no operators, have to use .set/.test methods).

(2nd attempt at actually commenting because the first one for some reason didn't show up in the thing...)

Reply:
Hah!  I considered that but having seen that Read=0 and Write=1 I assumed that they were never meant to be OR'd.  I didn't look deeper into Flags<> which is... pretty wild!  I'll amend.

Reply:
Hah! I considered that but having seen that Read=0 and Write=1 I assumed that they were never meant to be OR'd. I didn't look deeper into Flags<> which is... pretty wild! I'll amend.

Reply:
done!  it made some other stuff simpler too.